scripts/pip_audit_gate.py

```
@@ -0,0 +1,180 @@
+    stale_entries: list[str] = field(default_factory=list)
+    unused_entries: list[str] = field(default_factory=list)
+
+
+def load_allowlist(path: Path) -> list[AllowlistEntry]:
+    """Parse the YAML allowlist into typed entries.
+
+    Raises ValueError if the file is malformed or a required field is missing.
+    """
+    raw = yaml.safe_load(path.read_text())
+    if not isinstance(raw, dict) or "allowlist" not in raw:
+        raise ValueError(f"{path}: missing top-level 'allowlist' key")
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)